rest-new/src/main/java/org/jboss/pnc/rest/endpoints/AbstractEndpoint.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.rest.endpoints;
+
+import org.jboss.pnc.dto.DTOEntity;
+
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.facade.providers.api.Provider;
+import org.jboss.pnc.facade.validation.ConflictedEntryException;
+import org.jboss.pnc.rest.configuration.SwaggerConstants;
+import org.jboss.pnc.facade.validation.DTOValidationException;
+import org.jboss.pnc.rest.api.parameters.PageParameters;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+
+public class AbstractEndpoint<DTO extends REF, REF extends DTOEntity> {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final Provider<?, DTO, REF> provider;
+
+    private final Class<DTO> dtoClass;
+
+
+    public AbstractEndpoint(Provider<?, DTO, REF> provider, Class<DTO> dtoClass) {
+        this.provider = provider;
+        this.dtoClass = dtoClass;
+    }
+
+    protected DTO create(DTO dto) {
+        logger.debug("Creating an entity with body: " + dto.toString());
+        try{
+            dto = provider.store(dto);
+        } catch (ConflictedEntryException e) {
+            logger.debug("There was a conflict while creating entity: " + dto);
+            throw new WebApplicationException("There was a conflict while creating entity: " + dto, e, Response.status(Response.Status.CONFLICT).build());
+        } catch (DTOValidationException e) {
+            logger.debug("Validation error while creating entity: " + dtoClass.getSimpleName() + " with body: " + dto.toString());
+            throw new BadRequestException("Validation error while creating entity: " + dtoClass.getSimpleName() + " with body: " + dto.toString(), e);
+        }
```

Comment:
DTOValidationException should be made RuntimeException and not be catched. (also other places)